substrate-node/pallets/pallet-tfgrid/src/geo.rs

```diff
@@ -0,0 +1,24 @@
montovo","Kamen'-na-Obi","Kuz'minka","Sokolovo","Apazovo","Naberezhnyye Chelny","Lebedyan'","Yalta","Volna","Noginskaya","Lermontov","Volochek","Pavlovskaya Sloboda","Cherkizovo","Usman'","Tolstoy","Birzha","Valentin","Mor'ye","Dubrovka","Divo","Verkhotur'ye","Bulgakov","Bogdanovich","Maykor","Tazovskiy","Filatovskoye","Kasli","Chernushka","Muravlenko","Panayevsk","Tayshet","Zhuravna","Tulun","Medvenka","Tim","Granit","Malino","Semyonov","Kamennogorsk","Poztykeres","Medvedevo","Dubovka","Nevel'sk","Zhirnovsk","Yakovlevo","Tsudakhar","Ust'-Ilimsk","Volkhov","Syas'stroy","Aleksandrovka","Taldom","Timiryazev","Neva","Kamensk","Sven'","Beryozovsky","Gubkin","Pushchino","Volodarskiy","Salym","Karabash","Yemel'yanovo","Rosa","Trubchevsk","Katav-Ivanovsk","Krasnoznamensk","Svetlaya","Ostrov","Buret'","Kirilly","Ostankino","Sodyshka","Karla Marksa","Pikalyovo","Slavgorod","Solnechnodol'sk","Severo","Malysheva","Shumerlya","Aleksino","Skat","Gorbatov","Kel'dysh","Neftekumsk","Vikhorevka","Tapkhar"],"France":["Pont-Saint-Martin","Luce","Montaigu","La Chapelle-de-Brain","Saint-Mars-du-Desert","Challans","Le Mans","Saint-Laurent-de-la-Plaine","Baule","Cholet","Bouaye","Le Poire-sur-Velluire","Nantes","Trelaze","Pornichet","Sallertaine","La Montagne","Angers","Saint-Nazaire","Chateaubriant","Saint-Michel-le-Cloucq","Brissac-Quince","Saint-Julien-de-Concelles","Andouille","Bessay","Chavagnes-en-Paillers","Moutiers-les-Mauxfaits","Sainte-Luce-sur-Loire","Coueron","Savenay","Craon","Vritz","Chanteloup-les-Bois","Thouare-sur-Loire","Rouillon","Le Pellerin","Clisson","Saint-Herblain","Les Sables","Aizenay","Ecommoy","Saint-Hilaire-de-Chaleons","Brain-sur-Allonnes","Avrille","Breze","Doue-la-Fontaine","Coulaines","Pere","Reaumur","Olonne-sur-Mer","Montournais","Vertou","Sainte-Reine-de-Bretagne","Orvault","Evron","Parigne-l'Eveque","Juigne-sur-Sarthe","Chemille-Melay","Renaze","Aigrefeuille-sur-Maine","Pornic","Montbert","La Garnache","Champ","La Ferriere","Combree","Maze","Bouguenais","Port-Brillet","Saint-Christophe-en-Boucherie","Yzernay","La Meilleraye-de-Bretagne","Fontenay-le-Comte","La Plane","Treillieres","Saint-Saturnin-sur-Loire","Le Pouliguen","Saumur","Le Perrier","Blain","Bouchemaine","Loire","Le Lude","Mouzillon","Cande","La Faute-sur-Mer","Nort-sur-Erdre","Indre","Vallet","Carquefou","Saint-Etienne-de-Montluc","Rezé","Pipriac","Le Loroux","Lege","Quelaines-Saint-Gault","Pontchateau","Pouance","Laval","Saint-Pere-en-Retz","Torce-en-Vallee","Saint-Brevin-les-Pins","Grand","Le Pin","Saint-Berthevin","Suce-sur-Erdre","Argentre","Saint-Macaire-en-Mauges","Chavagnes-les-Redoux","Couesmes-Vauce","Loiron","Ruille-le-Gravelais","La Chapelle-aux-Lys","Saint-Fort","Chemaze","Le Coudray","Palluau","Ecuille","Saint-Barthelemy-d'Anjou","Challain-la-Potherie","Bazoges-en-Paillers","La Bruffiere","Saligny","Segre","Thorigne-d'Anjou","Soulvache","Grazay","Drain","Saint-Lambert-la-Potherie","Herbignac","Beaufou","Saint-Hilaire-de-Talmont","Lamnay","Change","La Chartre
```

Comment:
What kind of struct you are thinking about?

Meanwhile I investigated a bit and it is possible to use `str` in `no_std` env
The difference is that in the `no_std` context it refers to `core::primitive::str` instead of `std::str`

See here https://docs.rust-embedded.org/book/intro/no-std.html#summary for more details
> #![no_std] is a crate-level attribute that indicates that the crate will link to the core-crate instead of the std-crate.

Reply:
Vector of 

````
struct CountryInfo {
   Country: Vec<u8>,
   Regions: Vec<Vec<u8>>
}
````

?

Reply:
I see
Problem is that this string is used to be deserialized using serd_json:
~~~rust    
let data: Value = serde_json::from_str(country_city_str).unwrap();
~~~    
other option is dealing with slice `&[u8]` using `serde_json::from_slice()` if you think it s better
but I don t see real gain